src/CoreBundle/private/library/classes/TCMSResourceCollection.class.php

```diff
@@ -31,20 +32,40 @@ class TCMSResourceCollection
     protected $sCurrentCSSPath = null;
 
     /**
-     * Check if resource collection is enabled in config.
-     *
-     * @return bool
+     * @var IPkgCmsFileManager
+     */
+    private $cmsFileManager;
+
+    /**
+     * @var PortalDomainServiceInterface
+     */
+    private $portalDomainService;
+
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    public function __construct(
+        IPkgCmsFileManager $cmsFileManager,
```

Comment:
Changing constructor arguments is a BC break (the class didn't implement an interface before). We can remedy this by making arguments optional, similar to e.g. `TMapper_ViewPortSwitch`.

Reply:
ok

Does that work there? I would expect the type to begin with ? -> ?ActivePageServiceInterface

Reply:
`?`is not required, because null as default value automatically allows null values. But it doesn't hurt either, so it's fine as is now.